### PR TITLE
fix: select default value validation

### LINKED ISSUE #15701: [Bug]: Select widget's default value field throws error
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

For select widget, the error is displayed in default value field "Default value is missing in options. Please update the value.". Even though the value is provided and correct in the options. 

Note: This is observed in existing apps

Screenshot 2022-08-04 at 1 08 37 PM

### Steps To Reproduce

1. Open any existing app which has select widget
2. Observe the error is shown

### Public Sample App

_No response_

### Version

Release

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description

Default Value validation error.
The issue here is that options is still a string when defaultValue is validated
Here's a screenshot

Screenshot 2022-08-08 at 13 20 46

Fixes #15701 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/widgets/MultiSelectWidgetV2/widget/index.tsx [hunk 1]
```diff
@@ -144,6 +144,21 @@ export function defaultOptionValueValidation(
   }
 
   if (isValid && !_.isNil(parsed) && !_.isEmpty(parsed)) {
+    if (!Array.isArray(options)) {
+      if (typeof options === "string") {
```
Comment (sbalaji1192) [inline app/client/src/widgets/MultiSelectWidgetV2/widget/index.tsx:148]: club these two if together

File: app/client/src/widgets/SelectWidget/widget/index.tsx [hunk 2]
```diff
@@ -76,6 +75,20 @@ export function defaultOptionValueValidation(
   }
 
   if (isValid && !_.isNil(parsed) && parsed !== "") {
+    if (!Array.isArray(options)) {
```
Comment (sbalaji1192) [inline app/client/src/widgets/SelectWidget/widget/index.tsx:78]: here too

File: app/client/src/widgets/MultiSelectWidgetV2/widget/index.tsx [hunk 3]
```diff
@@ -144,6 +144,19 @@ export function defaultOptionValueValidation(
   }
 
   if (isValid && !_.isNil(parsed) && !_.isEmpty(parsed)) {
+    if (!Array.isArray(options) && typeof options === "string") {
```
Comment (ashit-rath) [inline app/client/src/widgets/MultiSelectWidgetV2/widget/index.tsx:147]: Is there a jest test case to validate this?

Comment (Tooluloope) [reply app/client/src/widgets/MultiSelectWidgetV2/widget/index.tsx:147]: Added

**Review by ashit-rath — DISMISSED** [review]

**Comment by chandannkumar** [comment]
Tested this PR on older apps to check the fix and it is working as expected

**Review by sbalaji1192 — DISMISSED** [review]

**Pull request merged**